drivers/hmis/spec/requests/hmis/custom_data_elements_spec.rb

```diff
@@ -65,16 +69,23 @@
       aggregate_failures 'checking response' do
         expect(response.status).to eq 200
         elements = result.dig('data', 'project', 'customDataElements')
-        expect(elements).to match([
-                                    a_hash_including(
-                                      'key' => cded1.key,
-                                      'label' => cded1.label,
-                                      'values' => [
-                                        a_hash_including('valueString' => cde1a.value_string),
-                                        a_hash_including('valueString' => cde1b.value_string),
-                                      ],
-                                    ),
-                                  ])
+        expect(elements).to match( # use match instead of contain_exactly here, the sort order should be deterministic
```

Comment:
I don't think this is ideal because it will only fail _sometimes_. It doesn't feel like it's truly testing the behavior. Instead, you could test that the first element is the one with the highest id (or lowest, whichever it is..).